src/appleseed/renderer/modeling/surfaceshader/physicalsurfaceshader.cpp

```diff
@@ -113,18 +113,32 @@ namespace
             // Compute lighting.
             AOVComponents components;
             ShadingComponents radiance;
-            for (size_t i = 0, e = m_lighting_samples; i < e; ++i)
-            {
-                shading_context.get_lighting_engine()->compute_lighting(
-                    sampling_context,
-                    pixel_context,
-                    shading_context,
-                    shading_point,
-                    radiance,
-                    components);
-            }
+
+            const Basis3d basis = shading_point.get_shading_basis();
+            shading_context.get_lighting_engine()->compute_lighting(
+                sampling_context,
+                pixel_context,
+                shading_context,
+                shading_point,
+                radiance,
+                components);
+
             if (m_lighting_samples > 1)
+            {
+                for (size_t i = 1, e = m_lighting_samples; i < e; ++i)
+                {
+                    shading_point.set_shading_basis(basis);
```

Comment:
Maybe add a short comment explaining that the shading basis stored in the shading point gets modified under certain circumstances?